azurerm/internal/services/loadbalancer/backend_address_pool_data_source.go

```diff
@@ -34,6 +35,34 @@ func dataSourceArmLoadBalancerBackendAddressPool() *schema.Resource {
 				ValidateFunc: validate.LoadBalancerID,
 			},
 
+			"backend_address": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"virtual_network_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"ip_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"network_interface_ip_configuration": {
```

Comment:
Is this an id? so should be
```suggestion
						"network_interface_ip_configuration_id": {
```

Reply:
I removed this one as it seems have already been covered by `backend_ip_configurations`. When users associate an IP to the LB BAP, this property will be empty for the `backend_address`. It is only set when users associate an NIC to the LB BAP, in which case since there is only one NIC can be attached in one time, the `backend_ip_configurations` is enough to query the info.